generator/src/main/java/com/graphicsfuzz/generator/transformation/donation/MakeArrayAccessesInBounds.java

```diff
@@ -87,11 +105,17 @@ private static boolean isArrayVectorOrMatrix(Type type) {
   }
 
   private static boolean staticallyInBounds(Expr index, Type type) {
-    if (!(index instanceof IntConstantExpr)) {
+    if (!(index instanceof IntConstantExpr || index instanceof UIntConstantExpr)) {
       return false;
     }
-    Integer indexValue = Integer.parseInt(((IntConstantExpr) index).getValue());
-    return indexValue >= 0 && indexValue < getSize(type);
+    Integer indexValue;
+    if (index instanceof IntConstantExpr) {
+      indexValue = Integer.parseInt(((IntConstantExpr) index).getValue());
+      return indexValue >= 0 && indexValue < getSize(type);
+    } else { // index instanceof UIntConstantExpr
```

Comment:
Because "getValue" returns something with "u" at the end of it, this would cause a NumberFormatException in the case of an in-bounds uint array access.

[This patch](https://github.com/google/graphicsfuzz/files/3498267/patch.txt) adds a test to expose that problem and uses getNumericValue to overcome it.

Reply:
Applied the patch.